pyqtgraph/Transform3D.py

```diff
@@ -45,8 +47,11 @@ def map(self, obj):
             v = QtGui.QMatrix4x4.map(self, Vector(obj))
             return type(obj)([v.x(), v.y(), v.z()])[:len(obj)]
         else:
-            return QtGui.QMatrix4x4.map(self, obj)
-            
+            retval = QtGui.QMatrix4x4.map(self, obj)
+            if not isinstance(retval, type(obj)):
```

Comment:
is `isinstance` what's wanted here? wouldn't that capture subclasses when we don't want them to?

For example, if `type(obj)`was of type `pg.Point`, wouldn't this return a `QPointF`?

(haven't actually tested this, just speculating)

Reply:
It would fail in the other direction. If `.map` returns a _subclass_ of the `obj(type)`, then it would pass this test and not try to downgrade the object into the less specific `obj(type)`, which seems like the correct behavior. So if we passed in a `QPoint` and it spat out a `QPointF`, we would still return the `QPointF`.

While I'm parameterizing this test, I can add those types, too.